src/libs/GooglePlacesUtils.ts

```diff
@@ -0,0 +1,53 @@
+type AddressComponent = {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    long_name: string;
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    short_name: string;
+    types: string[];
+};
+type FieldsToExtract = Record<string, Exclude<keyof AddressComponent, 'types'>>;
+
+/**
+ * Finds an address component by type, and returns the value associated to key. Each address component object
+ * inside the addressComponents array has the following structure:
+ * [{
+ *   long_name: "New York",
+ *   short_name: "New York",
+ *   types: [ "locality", "political" ]
+ * }]
+ */
+function getAddressComponents(addressComponents: AddressComponent[], fieldsToExtract: FieldsToExtract): Record<string, string> {
+    const result: Record<string, string> = {};
+    Object.keys(fieldsToExtract).forEach((key) => (result[key] = ''));
+
+    addressComponents.forEach((addressComponent) => {
+        addressComponent.types.forEach((addressType) => {
+            if (!(addressType in fieldsToExtract) || !(addressType in result && result[addressType] === '')) {
+                return;
+            }
+            const value = addressComponent[fieldsToExtract[addressType]] ? addressComponent[fieldsToExtract[addressType]] : '';
+            result[addressType] = value;
+        });
+    });
+    return result;
+}
+
+type AddressTerm = {value: string};
+type GetPlaceAutocompleteTermsResultKeys = 'country' | 'state' | 'city' | 'street';
```

Comment:
I think plural types are for arrays
```suggestion
type GetPlaceAutocompleteTermsResultKey = 'country' | 'state' | 'city' | 'street';
```

Reply:
My mistake, should be ok now @ArekChr